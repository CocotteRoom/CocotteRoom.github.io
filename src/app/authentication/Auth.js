import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import firebase from '../firebase';

export class Auth extends PureComponent {
  constructor(props) {
    super(props);
    this.handleSignIn = this.handleSignIn.bind(this);
  }

  async handleSignIn() {
    const { provider, onUserConnected, onError } = this.props;

    try {
      const {
        user,
        credential: { accessToken },
      } = await firebase.auth().signInWithPopup(provider);

      onUserConnected({ user, token: accessToken });
    } catch (e) {
      onError(e);
    }
  }

  render() {
    const { SigninButton } = this.props;

    return (
      <button onClick={this.handleSignIn}>
        <SigninButton />
      </button>
    );
  }
}

Auth.propTypes = {
  provider: PropTypes.shape({}).isRequired,
  SigninButton: PropTypes.func.isRequired,
  onUserConnected: PropTypes.func.isRequired,
  onError: PropTypes.func.isRequired,
};
