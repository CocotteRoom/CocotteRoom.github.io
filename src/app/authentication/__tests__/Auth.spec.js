import React from 'react';
import { shallow } from 'enzyme';
import { Auth } from '../Auth';

jest.mock('../../firebase', () => ({
  auth: jest.fn(() => ({
    signInWithPopup: jest.fn(() =>
      Promise.resolve({
        user: {},
        credential: { accessToken: 'token' },
      })),
  })),
}));

describe('Auth', () => {
  let props;

  const getWrapper = () => shallow(<Auth {...props} />);

  beforeEach(() => {
    props = {
      onUserConnected: jest.fn(),
      provider: {},
      SigninButton: () => <div>Signin button</div>,
    };
  });

  it('should match snapshot', () => {
    expect(getWrapper()).toMatchSnapshot();
  });

  it('should have called the different firebase method for authentications', async () => {
    const instance = getWrapper().instance();

    await instance.handleSignIn();

    expect(props.onUserConnected).toHaveBeenCalledWith({
      user: {},
      token: 'token',
    });
  });
});
