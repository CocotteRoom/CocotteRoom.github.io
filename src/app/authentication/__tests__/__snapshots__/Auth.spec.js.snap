// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth should match snapshot 1`] = `
<button
  onClick={[Function]}
>
  <SigninButton />
</button>
`;
